models/yang/sonic/sonic-nat.yang

```diff
@@ -0,0 +1,218 @@
+                    mandatory true;
+                    type uint16;
+                }
+
+                leaf nat_type {
+                    type enumeration {
+                        enum SNAT;
+                        enum DNAT;
+                    }
+                    default DNAT;
+                }
+
+                leaf twice_nat_id {
+                    type uint16 {
+                        range "1..9999" {
+                            error-message "Invalid Group Id for twice NAPT.";
+                        }
+                    }
+                }
+            }
+        }
+
+        container STATIC_NAT {
+            list STATIC_NAT_LIST {
+                key "global_ip";
+
+                leaf global_ip {
+                    type inet:ipv4-address;
+                }
+
+                leaf local_ip {
+                    mandatory true;
+                    type inet:ipv4-address;
+                }
+
+                leaf nat_type {
+                    type enumeration {
+                        enum SNAT;
+                        enum DNAT;
+                    }
+                    default DNAT;
+                }
+
+                leaf twice_nat_id {
+                    type uint16 {
+                        range "1..9999" {
+                            error-message "Invalid Group Id for twice NAT.";
+                        }
+                    }
+                }
+            }
+        }
+        container NAT_GLOBAL {
+            list NAT_GLOBAL_LIST {
+                key "key_leaf";
+                leaf key_leaf {
+                    type enumeration {
+                        enum Values;
+                    }
+                }
+                leaf admin_mode {
+                    type enumeration {
+                        enum ENABLE;
+                        enum DISABLE;
+                    }
+                }
+
+                leaf nat_timeout {
+                    type uint32 {
+                        range "300..432000" {
+                            error-message "Invalid NAT Timeout, valid range 300 sec to 432000 sec.";
+                        }
+                    }
+                }
+                leaf nat_tcp_timeout {
+                    type uint32 {
+                        range "300..432000" {
+                            error-message "Invalid NAT TCP Timeout, valid range 300 sec to 432000 sec.";
+                        }
+                    }
+                }
+                leaf nat_udp_timeout {
+                    type uint16 {
+                        range "120..600" {
+                            error-message "Invalid NAT UDP Timeout, valid range 120 sec to 600 sec.";
+                        }
+                    }
+                }
+            }
+        }
+        container NAT_POOL {
+            list NAT_POOL_LIST {
+                key "pool_name";
+
+                leaf pool_name {
+                    type string;
```

Comment:
Any pattern or length constraint for key ?

Reply:
Addressed